packages/react-input/src/components/Input/useInputStyles.ts

```diff
@@ -1,28 +1,225 @@
+    padding: `0 ${horizontalSpacing.sNudge}`,
+  }),
+  inputFilledDarker: theme => ({
+    backgroundColor: backgroundColors.filledDarker(theme),
+  }),
+  inputUnderline: theme => ({
+    backgroundColor: backgroundColors.transparent(theme),
+  }),
+  inputDisabled: theme => ({
+    color: theme.alias.color.neutral.neutralForegroundDisabled,
+    backgroundColor: theme.alias.color.neutral.transparentBackground,
+    cursor: 'not-allowed',
+    '::placeholder': {
+      color: theme.alias.color.neutral.neutralForegroundDisabled,
+    },
+  }),
+  inputWrapper: theme => ({
+    flexGrow: 1,
+    display: 'flex',
+    alignItems: 'center',
+    gap: horizontalSpacing.xxs,
+    padding: `0 ${horizontalSpacing.mNudge}`,
+
+    // Set this on inputWrapper rather than root since bookends will have separate background colors and borders
+    background: backgroundColors.filledLighter(theme), // for outline/filledLighter
+
+    // This may need to be conditionally applied to start/end corners when bookend styling is added
+    borderRadius: borderRadius(theme),
+  }),
+  inputWrapperSmall: {
+    padding: `0 ${horizontalSpacing.sNudge}`,
+  },
+  inputWrapperLarge: {
+    padding: `0 ${horizontalSpacing.m}`,
+  },
+  inputWrapperFilled: theme => ({
+    // This is set on inputWrapper since bookends have different borders.
+    // Don't set it in the main inputWrapper style to avoid override issues for underline appearance.
+    border: `1px solid ${theme.alias.color.neutral.transparentStroke}`,
+  }),
+  inputWrapperFilledDarker: theme => ({
+    backgroundColor: backgroundColors.filledDarker(theme),
+  }),
+  inputWrapperUnderline: theme => ({
+    backgroundColor: backgroundColors.transparent(theme),
+    border: 'none', // set later
+    borderRadius: 0, // corners look strange if rounded
+  }),
+  inputWrapperOutline: theme => ({
+    border: `1px solid ${theme.alias.color.neutral.neutralStroke1}`,
+  }),
+  /** bottom border for underline/outline appearances */
+  inputWrapperBorderBottom: theme => ({
+    borderBottom: `1px solid ${theme.alias.color.neutral.neutralStrokeAccessible}`,
+  }),
+  inputWrapperDisabled: theme => ({
+    border: `1px solid ${theme.alias.color.neutral.neutralStrokeDisabled}`,
+    borderRadius: borderRadius(theme), // because underline doesn't usually have a radius
+  }),
+  inside: theme => ({
+    color: theme.alias.color.neutral.neutralForeground3, // "icon color" in design spec
+    // special case styling for icons (most common case) to ensure they're centered vertically
+    '> span > svg': { display: 'block' },
+  }),
+  insideDisabled: theme => ({
+    color: theme.alias.color.neutral.neutralForegroundDisabled,
+  }),
 });
 
 /**
  * Apply styling to the Input slots based on the state
  */
 export const useInputStyles = (state: InputState): InputState => {
+  const { size, appearance = 'outline' } = state;
+  const small = size === 'small';
```

Comment:
Idea:  You could have a single object called something like this and only declare a single object.
`const styleFlags = { small: size === 'small', ... }`

Reply:
I thought about this a bit more and decided to split the styles into separate objects for most of the slots, which will probably be easier to maintain and allows accessing size and appearance styles by name.